### ISSUE TITLE
I think i bricked my camera, what can i do?

### ISSUE THREAD

**Issue by misaalanshori** [bcaller/v380-ipcam-firmware-patch#6]
I think I just bricked my camera by installing an old firmware I downloaded from a forum. the startup process ends with a segmentation fault. And it seems like it doesn't load the updates from the sd card anymore. is this camera just fully bricked now? are there any possible way to recover it?
the firmware originally was V380E12_WF8_PTZ_20180628 (as reported by the mobile app) and then i installed an update called AK3918E-V200_V.2.5.9.5 and now it wont start at all :(
any recommendation on what i should do? or is it just screwed? Does anyone have the latest firmware files for this camera?

**Comment by Arkady23** [comment]
You can not install early patches on these cameras, because each update is not a firmware, but a patch that fixes something in the previous firmware. If you have done so, then it is very difficult to fix the camera. Probably need to use the serial port. Solder and disassemble. And you can't do it with a simple soldering iron.

**Comment by misaalanshori** [comment]
Yeah, i think i really screwed up. I already have serial port access. But I still don't know the root password so I cant try to fix it this way. I don't think there is a way for me to fix this apart from just taking the flash out and flashing the correct firmware on it, so basically impossible...

**Comment by Arkady23** [comment]
I found information from someone who knows how to restore cameras. He writes that the program and the CH341A programmer are needed. At the same time, it attaches this file. But it is not known whether it is suitable for all or only for certain models.
[win25q64jvs10_my_dump_reset.zip](https://github.com/bcaller/v380-ipcam-firmware-patch/files/6565628/win25q64jvs10_my_dump_reset.zip)

**Comment by misaalanshori** [comment]
I'm guessing ill have to desolder the flash for this? also is there any info for what camera that dump is from?

**Comment by Arkady23** [comment]
The one who was able to restore it had a V380E2_C camera.

**Comment by misaalanshori** [comment]
okay i _think_ that is the exact same camera as mine. I guess ill have to buy the ch341a programmer first...
 edit: I think mine is actually V380E2_C2, I don't know if it has the same hardware or not... hopefully yes...

**Comment by Arkady23** [comment]
My camera is V380E2_C2. But I do not know how to make a dump. I have telnet. I can copy something. If you publish the firmware that you installed. I can tell you exactly what your model is.

**Comment by Arkady23** [comment]
I have telnet. I can copy something.

**Comment by Arkady23** [comment]
If you publish the firmware that you installed. I can tell you exactly what your model is.

**Comment by misaalanshori** [comment]
> My camera is V380E2_C2. But I do not know how to make a dump. 

Yeah, I think it requires you to desolder the flash and dumping the entire contents with a programmer. So its very difficult. I'm gonna try it with that dump anyways. maybe by some miracle it just works with my camera and I can get it to work again. if that doesn't work I might try modifying the current firmware by dumping it first.

i think I'm sure the model is V380E2_C2 because its in the patch headers. if you can, maybe try dumping all the partitions to a microsd card. not exactly sure how to do that tho...

**Comment by Arkady23** [comment]
I read that the dump is received like this:
`tar -cvf dump_V380E2_C2 /mnt/`
Do it?

**Comment by misaalanshori** [comment]
I'm not sure, i think that command basically dumps the /mnt/ folder, what is in it?

**Comment by Arkady23** [comment]
```
~ # ls
bin   dev   etc   ext   init  lib   mnt   mvs   proc  sbin  sys   tmp   usr   var
~ #
~ # cd /mnt
/mnt # ls
mtd     nand    sdcard
/mnt # cd mtd
/mnt/mtd # ls
DDNSClient.ini          dns_last_serverips.ini  mvconf                  recorde.ini             vg_boot.sh
ResetBind.ini           h265_1920x1080.cfg      mvsound                 recorder                vsipbroadcast
as9ipcwatchdog          idcheck.sh              network_Info.ini        resolv.conf             wificonf
as9nvserver             inet.conf               nvconfig.ini            stopallapp.sh
as9updatednsip          ipsourceserver.ini      nvipcstart.sh           styleId
audiofile_player        isp_gc1054.conf         patchrule.sh            user_info.ini
authority.ini           mtd_remount.sh          prerun                  venc.cfg
```

**Comment by misaalanshori** [comment]
hmm, maybe try dumping files in /dev/ instead? i think that would have the partitions.
can you list /dev/?

**Comment by Arkady23** [comment]
[dev_V380E2_C2.zip](https://github.com/bcaller/v380-ipcam-firmware-patch/files/6565680/dev_V380E2_C2.zip)

**Comment by misaalanshori** [comment]
hmm it looks like using tar might not be the way to go, all the devices are 0 bytes. i think you'd have to manually read each devices using cat and save it to the microsd, for example "cat /dev/mtd0 > /mnt/sdcard/mtd0" which will read from the mtd0 device and write it to the sdcard. But i dont even have a plan on how i would write these files to the camera flash, so it might end up being useless.

But, can you show me the output of "df"? to show the partition list

**Comment by Arkady23** [comment]
```
~ # df
Filesystem           1K-blocks      Used Available Use% Mounted on
/dev/root                 1280      1280         0 100% /
tmpfs                     1536       832       704  54% /mnt/mtd
/dev/mtdblock5            1536      1536         0 100% /usr
/dev/mtdblock6            1152      1152         0 100% /mvs
/dev/mtdblock7             384       384         0 100% /ext
/dev/mtdblock8            1536       832       704  54% /mnt/mtd
/dev/loop0                  95         1        95   1% /tmp/ramdisk
/dev/mmcblk0p1        65501952   1183392  64318560   2% /mnt/sdcard
~ #
```

**Comment by misaalanshori** [comment]
okay thanks. im still not 100% sure what to do next... Im currently in the process of bruteforcing the password but i doubt ill get anywhere with that. I guess My plan is going to be trying the V380E2_C on my camera and hoping it works (after dumping my current flash)

**Comment by Arkady23** [comment]
If it works, write it. Then I'll make you an installer for the web server.

**Comment by misaalanshori** [comment]
Okay, i already ordered the programmer. hopefully, i can get the firmware to work. it looks like it uses the same cpu (or at least brand of cpu) so hopefully it will be mostly compatible

I just bought the camera too! i was excited, but a series of bad decision brought me here

**Comment by misaalanshori** [comment]
I looked at the board and I think I found the flash. A 64mbit 8MB flash 25l6436f. I also found someone else's thread with a camera that uses the same flash and they had success but with a second camera to dump the flash image. so hopefully I can get it to work too...

i wish I didn't make a hole for the serial pins on the camera, I probably could've just returned it haha

**Comment by misaalanshori** [comment]
I have received the programmer and flashed the V380E2_C dump but unfortunately it failed to initialize everything. So its not compatible at all. But I am currently waiting for another camera from the same seller and hopefully it's the exact same model...

**Comment by misaalanshori** [comment]
I have received my second camera, and unfortunately it's a different model... only slightly different but different enough to make the firmware incompatible. the first one I noticed is that the SPI Flash and wifi modules are different.

I still tried dumping it and flashing it to the first camera and it does not work. But it is a different error, I uploaded a full log here: https://pastebin.com/J820nCSJ and it looks kinda similar to the logs of the camera before I installed the old problematic patch files. So there is some hope to maybe patching it somehow, but I don't know... Another thing that gave me hope is that the speakers and motors are working, and it looks like there is no compatibility error like when I tried the V380E2_C dump.

The booting process is basically stuck at this part, it just continuously loops at this and doesn't continue. While I do not have the boot logs before i corrupted the old camera, it does looks very similar. Especially the "DNS parsing error", i remember seeing that on the original not yet fully corrupted firmware.
looping part of the log:
```
=============================================================
as9updatednsip ver: 8.5.9.1_20190413_9d4b238[build:Mar 30 2020_15:30:49]
=============================================================
 
Check encrypt calculate ID:147895838 MACID:-1 IPCID:45342452 incorrect! 
[0:01:12][1m[33m[WRN][ulog][TCP]DNS parsing error, retcode=[-23][0m
[0:01:12][1m[33m[WRN][ulog][TCP]DNS parsing error, retcode=[-23][0m
[0:01:12][1m[33m[WRN][ulog]Upload device base information failed ! retcode=[-23][0m
updatednsip gethostbyname_r err.
updnsGetHost success 
gethost by name fail Domain:
time:312768054,Domain:47.107.163.76 
base64:JirpJYL8u9d3QstCXL4eGw==
before md5:encrypt=JirpJYL8u9d3QstCXL4eGw==&timestamp=312768054hsshop2016
{"encrypt":"JirpJYL8u9d3QstCXL4eGw==","timestamp":312768054,"sign":"cd4e55e3d1fb8326abf100791eb51102"}
http form post error= connect:: Network is unreachable
47.107.163.76:8002timestamp:0
usb 1-1: device descriptor read/64, error -110
[0:01:14][1m[33m[WRN][ulog][TCP]DNS parsing error, retcode=[-23][0m
[0:01:14][1m[33m[WRN][ulog][TCP]DNS parsing error, retcode=[-23][0m
[0:01:14][1m[33m[WRN][ulog]Upload device base information failed ! retcode=[-23][0m
updatednsip gethostbyname_r err.
updatednsip gethostbyname_r err.
[0:01:16][1m[33m[WRN][ulog][TCP]DNS parsing error, retcode=[-23][0m
[0:01:16][1m[33m[WRN][ulog][TCP]DNS parsing error, retcode=[-23][0m
[0:01:16][1m[33m[WRN][ulog]Upload device base information failed ! retcode=[-23][0m
exited, status=255
updatednsip_r Child PID is 577
```
So at this point, I think I'm gonna have to put this first camera on the shelf for now. Until someone with the exact same V380E2_C2 camera and a CH341A programmer can do a dump and upload them. At least I do have one working camera now...

I won't close this issue as this repo is one of the first search results you see when looking up the v380 patch so hopefully someone sees this and can help with providing a dump.

*Also, another thing I forgot to mention in this issue thread is that the camera was already boot looping even before I installed the old problematic patch, and I installed the patch with the hopes that it will somehow reset the entire board and fix everything. But instead, it made everything worse

I also just realized this might be the perfect opportunity to learn embedded Linux. I can try to compile my own Linux and make my own image for the board, and maybe figure out the drivers for video, USB, wifi, and everything. But that is definitely not going to be a quick project lol

**Comment by misaalanshori** [comment]
Also, Arkady23. Maybe if you're not busy and have some time and are comfortable with doing this, you can get the programmer and do a dump for me :). You don't need to desolder the flash, you can get the ch341a programmer and sop8 ic clip for less than $10. Of course, you don't have to do it if you don't want to, it's not very urgent haha

**Comment by Arkady23** [comment]
Do you know the HwName/Soft_id of the new camera?
If you are lucky enough to see a screen with information that there is a new firmware, then make sure to take a screenshot of this screen. It displays some of the information from the camera's HwName.

**Comment by misaalanshori** [comment]
Oh yeah, I was going to do that and forgot. the new camera currently has V380E12_WF6_PTZ_E2_C2_v2.6.5.9, and the old one has the exact same firmware name!!! so maybe the hardware differences didn't matter and it runs the same firmware. but that kinda made it even weirder because now it's probably not a firmware problem, but something else...

I guess now ill have to figure that out...

**Comment by misaalanshori** [comment]
after a long email conversation with the v380 email, they actually sent a firmware patch. so maybe it'll help

edit: It did not. but it think i might have figured out what the problem is:
`check encrypt calculate ID:147895838 MACID:-1 IPCID:45342452 incorrect!`
the IPCID is the number on the bottom of the camera and i have a feeling its what is being stored in the eeprom

**Comment by Arkady23** [comment]
You have V380E2_C2. Then install the web server https://github.com/Arkady23/V380-Web httpd_V380E2_C2_v20210530.zip
>they actually sent a firmware patch

it would be useful not only for you. You can post it to anyone who has a broken camera.

**Comment by misaalanshori** [comment]
unfortunately, the camera can't connect to the wifi, so I don't think I would be able to access the webserver. As for the new camera, I am afraid to do any mods to it as I actually need to use the camera, so I'm not taking any risks with the new one


here is the firmware patch they gave me [AK_3918E-V200__(for new sdk version above V2.6.2.0)_V2.6.6.6_universal_upgrade_patch_2021-03-25.zip](https://github.com/bcaller/v380-ipcam-firmware-patch/files/6602668/AK_3918E-V200__.for.new.sdk.version.above.V2.6.2.0._V2.6.6.6_universal_upgrade_patch_2021-03-25.zip)

**Comment by Arkady23** [comment]
the web server is installed **absolutely safely** on the sd-card.
>unfortunately, the camera can't connect to the wifi

You tell them that. Perhaps they will make a more global patch that will help you. If they send you more, put it here!

**Comment by Arkady23** [comment]
>unfortunately, the camera can't connect to the wifi

You tell them that. Perhaps they will make a more global patch that will help you.

**Comment by misaalanshori** [comment]
Yeah, I have sent a reply to it saying the patch did not work and that I'm still unable to do the wifi setup.

These, error messages are bizarre, why is it trying to do DNS stuff and connect to servers even though it hasn't even set up wifi yet... its getting very frustrating
```
=============================================================
as9updatednsip ver: 8.5.9.1_20190413_9d4b238[build:Mar 30 2020_15:30:49]
=============================================================

Check encrypt calculate ID:147895838 MACID:-1 IPCID:45342452 incorrect! 
[929]check wifi module (lsusb)
[0:15:28][1m[33m[WRN][ulog][TCP]DNS parsing error, retcode=[-23][0m
[0:15:28][1m[33m[WRN][ulog][TCP]DNS parsing error, retcode=[-23][0m
[0:15:28][1m[33m[WRN][ulog]Upload device base information failed ! retcode=[-23][0m
updatednsip gethostbyname_r err.
updnsGetHost success 
gethost by name fail Domain:
time:312768910,Domain:47.107.163.76 
base64:JirpJYL8u9d3QstCXL4eGw==
before md5:encrypt=JirpJYL8u9d3QstCXL4eGw==&timestamp=312768910hsshop2016
{"encrypt":"JirpJYL8u9d3QstCXL4eGw==","timestamp":312768910,"sign":"9121d3c4ff1be9ab30377f6f5833ea90"}
http form post error= connect:: Network is unreachable
47.107.163.76:8002timestamp:0
[0:15:30][1m[33m[WRN][push][TCP]DNS parsing error, retcode=[-23][0m
[0:15:30][1m[33m[WRN][push]Get alarm servers list from[alarmserverlist.av380.net:8888] failed, retcode=[-23][0m
[0:15:30][1m[33m[WRN][ulog][TCP]DNS parsing error, retcode=[-23][0m
[0:15:30][1m[33m[WRN][ulog][TCP]DNS parsing error, retcode=[-23][0m
[0:15:30][1m[33m[WRN][ulog]Upload device base information failed ! retcode=[-23][0m
[0:15:30][1m[33m[WRN][urec][TCP]DNS parsing error, retcode=[-23][0m
[0:15:30][1m[33m[WRN][urec]Connect [rapi.av380.net:8002] failed, retcode=[-23][0m
[0:15:30][1m[33m[WRN][urec][TCP]Connect erron=[101][0m
[0:15:30][1m[33m[WRN][urec][TCP]Connect [47.107.163.76:8002] failed !![0m
[0:15:30][1m[33m[WRN][urec]Connect [47.107.163.76:8002] failed, retcode=[-17][0m
[0:15:30][1m[33m[WRN][urec]Connect server failed[0m
updatednsip gethostbyname_r err.
updatednsip gethostbyname_r err.
[932]check wifi module (lsusb)
[0:15:32][1m[33m[WRN][ulog][TCP]DNS parsing error, retcode=[-23][0m
[0:15:32][1m[33m[WRN][ulog][TCP]DNS parsing error, retcode=[-23][0m
[0:15:32][1m[33m[WRN][ulog]Upload device base information failed ! retcode=[-23][0m
exited, status=255
[0:15:34][1m[33m[WRN][ulog][TCP]DNS parsing error, retcode=[-23][0m
updatednsip_r Child PID is 2481

```
maybe i need to try running my own scripts on it again and maybe i could find something there...

**Comment by Arkady23** [comment]
You need to look at your network_Info.ini file. If you can't do it. I can make you a patch that will copy to the sd-card folder /mnt/mtd, where this file is located and then it will be clear what you have there is corrupted. Camera data, its hwname, username, passwords, and other information. Everything is recorded in this folder. Most likely, some of your data is corrupted.

**Comment by misaalanshori** [comment]
But I'm running a fresh firmware sourced directly from a new camera. Why would it be corrupted? 

where is this network_info.ini file? is it in /mnt/mtd? i guess ill try to figure it out again tomorrow

**Comment by Arkady23** [comment]
Where do you buy a V380 camera? There are no such cameras where I bought them.

**Comment by misaalanshori** [comment]
I bought both of them from the same seller at a local online shop. 
Also, what do you mean by "There are no such cameras where I bought them." I thought we have the same camera model right?

But at this point, I have decided to go with making my own firmware instead. It seems like there will be some problems with the camera communication with the camera servers because of id mismatch and stuff. I might try to figure out how to get the original firmware working again, but I think this route is going to be more fun (especially now that I have most things figured out)

So far I have figured out how to modify the firmware directly and _**finally**_ getting a root shell:
```
V380E login: root
Password: 
welcome to file system
[root@V380E ~]$ ls
bin   dev   etc   ext   init  lib   mnt   mvs   proc  sbin  sys   tmp   usr   var
[root@V380E ~]$ 
```

**Comment by misaalanshori** [comment]
Actually, now that I have root access, fixing this might be as easy as modifying the DEV_ID on the network_Info.ini file you mentioned. So ill keep that in mind. I guess this might be something other people also need to know if they flash their camera with a firmware dump from a different camera. Luckily the DEV_ID is also on a label under the camera, so that's useful.

Also, here is the firmware dump for the V380E2_C2: [fwdump.zip](https://github.com/bcaller/v380-ipcam-firmware-patch/files/6621275/fwdump.zip)

**Comment by Arkady23** [comment]
> local online shop

Can I make an order from Russia in this shop?

**Comment by misaalanshori** [comment]
I don't think you can. Its https://lazada.co.id

Also, i got a bit stuck with the custom firmware. So i tried changing the DEV_ID on the original firmware, but it still doesn't work and gives the same error... 

So unfortunately i still have not found the source of error... That file might be part of it, but i think theres more problems...

**Comment by Arkady23** [comment]
> Check encrypt calculate ID:147895838 **MACID:-1** IPCID:45342452 incorrect!

Maybe you don't have a MAC address? By the way, it is also written in this network_Info.ini file.

**Comment by misaalanshori** [comment]
Yeah I checked the mac address and it's in that file. I'm not sure why it's
not detected there.

Also, can you send your boot logs? i think if you don't have serial access
you can see it from dmesg, but i'm not sure how much info that would have.
Maybe I could try to see any differences.

**Comment by Arkady23** [comment]
You have a second camera like this! I can only say that in the /tmp folder I see the wifimac.ini file with 3 lines:
```
[WIFIMAC]
MACID=12345678
mac_addr=1234567890AB
```
I replaced my numbers. There is no ": " in the MAC address.
I think this file is created when the camera is turned on, and then used to connect to WiFi.

**Comment by misaalanshori** [comment]
Hmm okay. I actually found another possible problem. While trying to get wifi to work manually from the root shell i noticed a bunch of USB errors, and after more looking around i noticed that the driver for the Wifi doesn't match the chip inside. The driver was for the atbm603x chip, but the chip i have is the atbm6022. Is it possible for you to send the driver files from your camera? Just copy the /ext and /mvs folder.
Thank you so much btw, you have helped so much with this!

**Comment by Arkady23** [comment]
[mvs_V380E2_C2.zip](https://github.com/bcaller/v380-ipcam-firmware-patch/files/6630878/mvs_V380E2_C2.zip)

**Comment by misaalanshori** [comment]
Thanks I will try getting that on the camera tomorrow. Though it looks like my wifi chip is different than yours. Your driver is for a Realtek Chip, but mine is an Altobeam chip: https://www.altobeam.com/en/contents/72/2.html. I guess they change out the wifi chip very often...

**Comment by drtanzil** [comment]
I have encountered the same problem.
Updated with old patch and boom!

But in my case device is turning on, it says..
_Device is starting up...
Device startup completed._

Then do nothing. It cannot establish a wifi access point.
May be the WIFI USB Driver is different.

Unfortunately I did not take any screenshot of Hdw & soft ID.
I'm in conversation with v380 team, has send them chip photo if they can send me proper patch.


![Board](https://user-images.githubusercontent.com/11825577/142032429-593a3dd4-0857-47d3-8b43-9fa60d226f17.jpg)

![unnamed](https://user-images.githubusercontent.com/11825577/142032327-e77e5ef1-fe1d-4d2c-806e-75f720034a1a.jpg)
![unnamed](https://user-images.githubusercontent.com/11825577/142032337-f1b08dcd-9738-4e49-8d31-ebccd55e0844.png)

**Comment by exploitwritter** [comment]
> 

Can someone help me with the root access password by UART of V380E?

**Comment by Geotv123** [comment]
> I bought both of them from the same seller at a local online shop. Also, what do you mean by "There are no such cameras where I bought them." I thought we have the same camera model right?
> 
> But at this point, I have decided to go with making my own firmware instead. It seems like there will be some problems with the camera communication with the camera servers because of id mismatch and stuff. I might try to figure out how to get the original firmware working again, but I think this route is going to be more fun (especially now that I have most things figured out)
> 
> So far I have figured out how to modify the firmware directly and _**finally**_ getting a root shell:
> 
> ```
> V380E login: root
> Password: 
> welcome to file system
> [root@V380E ~]$ ls
> bin   dev   etc   ext   init  lib   mnt   mvs   proc  sbin  sys   tmp   usr   var
> [root@V380E ~]$ 
> ```

Hi, I’m really hoping you can share how you modified the firmware directly to get root. I have dumped the flash. Thank you

**Comment by wlgallesio** [comment]
Good afternoon, maybe someone can help me, I have a v380 camera, I sent an email to v380technical@gmail.com asking to release the RTSP port, they sent me a file, when I put the file (SPECIAL_V380_AK_3918E-V330_20MB-DMA_V9.9.9.9_2021-09-24) on the micro SD and turned on the camera, then It didn't work anymore, the reset button doesn't work anymore, can anyone help me solve it? It's a new camera
Does anyone know where I can connect UART? and it would have the complete firmware

[SPECIAL_V380_AK_3918E-V330_20MB-DMA_V9.9.9.9_2021-09-24 (2).zip](https://github.com/bcaller/v380-ipcam-firmware-patch/files/15097340/SPECIAL_V380_AK_3918E-V330_20MB-DMA_V9.9.9.9_2021-09-24.2.zip)
![v380 modelo](https://github.com/bcaller/v380-ipcam-firmware-patch/assets/125309780/4e87c90d-2a7e-4641-a2f3-74362050fe22)
![v380 placa](https://github.com/bcaller/v380-ipcam-firmware-patch/assets/125309780/bb4b7a9e-9a1f-4ad2-aded-22f24cbe8b2a)

**Comment by wlgallesio** [comment]
I found tx and rx and got access, but I don't know how to fix it

![IMG_9331](https://github.com/bcaller/v380-ipcam-firmware-patch/assets/125309780/1e4bdab5-3622-49d4-9a64-fc898ad87f4f)
